src/pages/item/item-actions/ItemActions.tsx

```diff
@@ -21,29 +24,113 @@ export default function ItemActions(props: { itemId: string | number }) {
 				"Can't load Downstream References - data still loading or failed to do so."
 			);
 		}
+		
 	};
 
+	const showDependenciesHandler = async () => {
+
+		// loadCards
+		// if (data && !showDependenciesDisabled) {
+		// 	const ids = data.outgoingAssociations.map((d) =>
+		// 		d.itemRevision.id.toString()
+		// 	);
+		// 	setItemIds(ids);
+		// 	setQueryString(`item.id IN (${ids.join(',')})`);
+		// } else {
+		// 	console.warn(
+		// 		"Can't load Associations - data still loading or failed to do so."
+		// 	);
+		// }
+
+		if (data && !showDependenciesDisabled) {
+			const { itemIds, associationIds } = data.outgoingAssociations.reduce(
+				(acc, curr) => {
+					acc.itemIds.push(curr.itemRevision.id.toString());
+					acc.associationIds.push(curr.id.toString());
+					return acc;
+				},
+				{ itemIds: [], associationIds: [] }
+			);
+			setItemIds(itemIds);
+			
+		  
+			const startCardId = await findAppCardId(props.itemId.toString())
+		  
+			for (const [index, id] of itemIds.entries()) {
+				const endCardId = await findAppCardId(id);
+				
+				setAssociationId(associationIds[index])
+			
+				if (startCardId && endCardId) {
+					const connector = await miro.board.createConnector({
+					start: {
+						item: startCardId.toString()
+					},
+					end: {
+						item: endCardId.toString()
+					}
+					});
+					console.log(`Connector ${index} created between ${startCardId} and ${endCardId}:`, connector);
+				} else {
+					console.warn(`Could not find app_card for id ${index}: ${id}`);
+				}
+			}
+		  } else {
+			console.warn("Can't load Associations - data still loading or failed to do so.");
+		  }
+		  setAssociationId('')
+	};
```

Comment:
Here, you mean? @C2H6Ethan

Reply:
yes, the last thing I tried was updating a state variable to trigger a useEffect at line 62 but that also doesn't work.

Reply:
Regarding line 62; The `setState` hook isn't as straight forward as one would think/hope. Here, I think it will simply end up setting the association id to `''`, as done on line `82`. Because React optimizes updating the state and will, if you update it several times in the same function, only update it once - with the last value you give it.  

That behaviour would explain if you'd be getting an API error, for querying a relation with id `''`.  

But  if that's not it, you gotta provide some more intel on this. When do which errors appear, what's expected, what the actual behaviour, and so on.  
If you properly recondition your errors, you'll often untie the knots yourself in the first place. (Although here, if it's only `useState`'s special behaviour, it wouldn't be that likely/easy).

Reply:
do you know how I can call the associations query inside the for loop, how would it be possible, is there a different way